uaclient/files/state_files.py

```diff
@@ -232,3 +232,45 @@ def __init__(
 
 
 reboot_cmd_marker_file = UAFile("marker-reboot-cmds-required")
+
+
+class AnboxCloudCredentials(DataObject):
+    fields = [
+        Field("anbox_images_url", StringDataValue),
+        Field("anbox_images_resource_token", StringDataValue),
+        Field("anbox_cloud_apt_token", StringDataValue),
+        Field("anbox_cloud_apt_url", StringDataValue),
+    ]
+
+    def __init__(
+        self,
+        anbox_images_url: str,
+        anbox_images_resource_token: str,
+        anbox_cloud_apt_token: str,
+        anbox_cloud_apt_url: str,
+    ):
+        self.anbox_images_url = anbox_images_url
+        self.anbox_images_resource_token = anbox_images_resource_token
+        self.anbox_cloud_apt_token = anbox_cloud_apt_token
+        self.anbox_cloud_apt_url = anbox_cloud_apt_url
+
+
+class AnboxCloudData(DataObject):
+    fields = [
+        Field("anbox_attributes", AnboxCloudCredentials),
+    ]
+
+    def __init__(self, anbox_attributes: AnboxCloudCredentials):
+        self.anbox_attributes = anbox_attributes
+
+
+anbox_cloud_credentials_file = DataObjectFile(
+    AnboxCloudData,
+    UAFile(
+        "anbox-cloud-credentials",
+        defaults.DEFAULT_PRIVATE_DATA_DIR,
+        private=True,
+    ),
+    DataObjectFileFormat.JSON,
+    optional_type_errors_become_null=True,
```

Comment:
This only needs to be set to True if we expect other people or software to be modifying the file (which could result in a value of the wrong type getting written)